### PR TITLE
Incorporate Detekt for static code checks

### PR DESCRIPTION
**Pull request by Drakojin**
Ongoing pull request until all detekt findings are fixed, or the config is changed to our liking.

### CODE REVIEW

File: livingdoc-engine/src/main/kotlin/org/livingdoc/engine/execution/examples/decisiontables/DecisionTableFixtureChecker.kt [hunk 1]
```diff
@@ -40,10 +46,10 @@ internal object DecisionTableFixtureChecker {
         val errors = mutableListOf<String>()
         val knownAliases = mutableSetOf<String>()
 
-        val handleAlias: (String) -> Unit = {
-            if (knownAliases.contains(it))
-                errors.add("Alias <$it> is used multiple times!")
-            else knownAliases.add(it)
+        val handleAlias: (String) -> Unit = { string ->
```
Comment (slu-it) [inline livingdoc-engine/src/main/kotlin/org/livingdoc/engine/execution/examples/decisiontables/DecisionTableFixtureChecker.kt]: The parameter `string` should be named `alias`

File: livingdoc-engine/src/main/kotlin/org/livingdoc/engine/LivingDoc.kt [hunk 2]
```diff
@@ -77,34 +78,40 @@ private data class ExecutableDocumentModel(
...
-                    .also { require(it.size == 2) { "Illegal annotation value '${annotation.value}'." } }
+                .also { require(it.size == 2) { "Illegal annotation value '${annotation.value}'." } }
             return DocumentIdentifier(values[0], values[1])
         }
 
         private fun validate(document: Class<*>) {
             if (document.executableDocumentAnnotation == null) {
-                throw IllegalArgumentException("ExecutableDocument annotation is not present on class ${document.canonicalName}.")
+                throw IllegalArgumentException(
+                    "ExecutableDocument annotation is not present on class " +
```
Comment (slu-it) [inline livingdoc-engine/src/main/kotlin/org/livingdoc/engine/LivingDoc.kt]: might the lines 97 and 98 now fit in one line?

File: livingdoc-engine/src/main/kotlin/org/livingdoc/engine/execution/examples/decisiontables/DecisionTableFixtureChecker.kt [hunk 3]
```diff
@@ -59,8 +65,8 @@ internal object DecisionTableFixtureChecker {
         flatMapper: (T) -> Iterable<String>,
         handler: (String) -> Unit
     ) {
-        elements.forEach {
-            it.getAnnotationsByType(annotation.java)
+        elements.forEach { annotatedElement ->
```
Comment (slu-it) [inline livingdoc-engine/src/main/kotlin/org/livingdoc/engine/execution/examples/decisiontables/DecisionTableFixtureChecker.kt]: `element` might be enough in this context

File: livingdoc-engine/src/main/kotlin/org/livingdoc/engine/execution/examples/scenarios/matching/StepTemplate.kt [hunk 4]
```diff
@@ -65,20 +65,18 @@ internal class StepTemplate(
...
                     isVariable = true
                     if (lastIndex < i) {
                         tokens.add(templateAsString.substring(lastIndex, i))
                     }
                     lastIndex = i
                 } else if (c == '}' && !isEscaped(templateAsString, i)) {
                     if (!isVariable) {
-                        throw IllegalFormatException("Illegal closing curly brace at position $i!\nOffending string was: $templateAsString")
+                        throw IllegalFormatException(
+                            "Illegal closing curly brace at position $i!\n" +
```
Comment (slu-it) [inline livingdoc-engine/src/main/kotlin/org/livingdoc/engine/execution/examples/scenarios/matching/StepTemplate.kt]: might this fit in a single line now?

File: livingdoc-engine/src/main/kotlin/org/livingdoc/engine/execution/examples/scenarios/matching/StepTemplate.kt [hunk 5]
```diff
@@ -96,6 +94,27 @@ internal class StepTemplate(
...
 
+        private fun validateVariables(
+            isVariable: Boolean,
+            i: Int,
+            templateAsString: String,
+            isPreceededByVariable: Boolean
+        ) {
+            if (isVariable) {
+                throw IllegalFormatException(
+                    "Illegal opening curly brace at position $i!\n" +
```
Comment (slu-it) [inline livingdoc-engine/src/main/kotlin/org/livingdoc/engine/execution/examples/scenarios/matching/StepTemplate.kt]: single line?

File: livingdoc-engine/src/main/kotlin/org/livingdoc/engine/reporting/HtmlElements.kt [hunk 6]
```diff
@@ -33,11 +31,14 @@ class HtmlTable(val renderContext: HtmlRenderContext, val tableResult: Result, v
...
             val tableFailedRow = Element("tr")
             tableFailedRow.appendChild(
-                    Element("td").apply {
-                        setStyleClasses(HtmlReportTemplate.CSS_CLASS_BORDER_BLACK_ONEPX, determineCssClassForBackgroundColor(tableResult))
-                        attr("colspan", columnCount.toString())
-                        appendChild(createFailedPopupLink(renderContext, tableResult))
-                    })
+                Element("td").apply {
+                    setStyleClasses(
+                        HtmlReportTemplate.CSS_CLASS_BORDER_BLACK_ONEPX,
```
Comment (slu-it) [inline livingdoc-engine/src/main/kotlin/org/livingdoc/engine/reporting/HtmlElements.kt]: for better readability we could use parameter names here

File: livingdoc-engine/src/main/kotlin/org/livingdoc/engine/reporting/HtmlElements.kt [hunk 7]
```diff
@@ -65,10 +66,13 @@ fun HtmlTable.rows(rows: List<RowResult>) {
     fun appendCellToDisplayFailedRowIfNecessary(newRow: Element, rowResult: Result) {
         if (rowResult is Result.Failed || rowResult is Result.Exception) {
             newRow.appendChild(
-                    Element("td").apply {
-                        setStyleClasses(HtmlReportTemplate.CSS_CLASS_BORDER_BLACK_ONEPX, determineCssClassForBackgroundColor(rowResult))
-                        appendChild(createFailedPopupLink(htmlTable.renderContext, rowResult))
-                    })
+                Element("td").apply {
+                    setStyleClasses(
+                        HtmlReportTemplate.CSS_CLASS_BORDER_BLACK_ONEPX,
```
Comment (slu-it) [inline livingdoc-engine/src/main/kotlin/org/livingdoc/engine/reporting/HtmlElements.kt]: for better readability we could use parameter names here

File: livingdoc-engine/src/main/kotlin/org/livingdoc/engine/reporting/HtmlElements.kt [hunk 8]
```diff
@@ -100,9 +105,9 @@ fun HtmlTable.rows(rows: List<RowResult>) {
 
 private fun createFailedPopupLink(renderContext: HtmlRenderContext, result: Result): Element {
     fun createStacktrace(e: Throwable): String {
-        return StringWriter().use {
-            e.printStackTrace(PrintWriter(it))
-            it.toString()
+        return StringWriter().use { stringWriter ->
```
Comment (slu-it) [inline livingdoc-engine/src/main/kotlin/org/livingdoc/engine/reporting/HtmlElements.kt]: `stringWriter` >> `writer` ?

File: livingdoc-engine/src/main/kotlin/org/livingdoc/engine/reporting/HtmlReportTemplate.kt [hunk 9]
```diff
@@ -125,21 +125,21 @@ class HtmlReportTemplate {
     }
 
     private fun createErrorPopups(renderContext: HtmlRenderContext): String {
-        return renderContext.popupErrors.joinToString(separator = "") {
+        return renderContext.popupErrors.joinToString(separator = "") { htmlError ->
```
Comment (slu-it) [inline livingdoc-engine/src/main/kotlin/org/livingdoc/engine/reporting/HtmlReportTemplate.kt]: `htmlError` >> `error`

File: livingdoc-engine/src/main/kotlin/org/livingdoc/engine/execution/examples/decisiontables/DecisionTableFixtureChecker.kt (hunk 3 above)
Comment (Drakojin) [reply livingdoc-engine/src/main/kotlin/org/livingdoc/engine/execution/examples/decisiontables/DecisionTableFixtureChecker.kt]: we won't win anything with making it less declarative. No use in making the effort

File: livingdoc-engine/src/main/kotlin/org/livingdoc/engine/execution/examples/scenarios/matching/StepTemplate.kt (hunk 4 above)
Comment (Drakojin) [reply livingdoc-engine/src/main/kotlin/org/livingdoc/engine/execution/examples/scenarios/matching/StepTemplate.kt]: My suggestion would be raw string with trim margin. reflects the message better.

File: livingdoc-engine/src/main/kotlin/org/livingdoc/engine/reporting/HtmlElements.kt (hunk 7 above)
Comment (Drakojin) [reply livingdoc-engine/src/main/kotlin/org/livingdoc/engine/reporting/HtmlElements.kt]: I don't see any nameable parameters here ;)

File: livingdoc-engine/src/main/kotlin/org/livingdoc/engine/reporting/HtmlElements.kt (hunk 8 above)
Comment (Drakojin) [reply livingdoc-engine/src/main/kotlin/org/livingdoc/engine/reporting/HtmlElements.kt]: the block uses a printwriter -> stringwriter improves readability

**Pull request merged**